src/lib/ffi/ffi.h

```diff
@@ -2196,6 +2204,198 @@ int botan_x509_cert_verify(int* validation_result,
+int botan_x509_ext_as_blocks_get_rdi(botan_x509_ext_as_blocks_t as_blocks, int* present, size_t* count);
+
+BOTAN_FFI_EXPORT(3, 9)
+int botan_x509_ext_as_blocks_get_rdi_at(botan_x509_ext_as_blocks_t as_blocks, size_t i, uint32_t* min, uint32_t* max);
+
+/*
+* X.509 cert param builder
+*/
+typedef struct botan_x509_cert_params_builder_struct* botan_x509_cert_params_builder_t;
+
+BOTAN_FFI_EXPORT(3, 9) int botan_x509_cert_params_builder_destroy(botan_x509_cert_params_builder_t builder);
+
+BOTAN_FFI_EXPORT(3, 9)
+int botan_x509_create_cert_params_builder(botan_x509_cert_params_builder_t* builder_obj,
+                                          const char* opts,
+                                          uint32_t* expire_time);
+
+BOTAN_FFI_EXPORT(3, 9)
+int botan_x509_cert_params_builder_add_common_name(botan_x509_cert_params_builder_t builder, const char* name);
+
+BOTAN_FFI_EXPORT(3, 9)
+int botan_x509_cert_params_builder_add_country(botan_x509_cert_params_builder_t builder, const char* country);
+
+BOTAN_FFI_EXPORT(3, 9)
+int botan_x509_cert_params_builder_add_organization(botan_x509_cert_params_builder_t builder, const char* organization);
+
+BOTAN_FFI_EXPORT(3, 9)
+int botan_x509_cert_params_builder_add_org_unit(botan_x509_cert_params_builder_t builder, const char* org_unit);
+
+BOTAN_FFI_EXPORT(3, 9)
+int botan_x509_cert_params_builder_add_locality(botan_x509_cert_params_builder_t builder, const char* locality);
+
+BOTAN_FFI_EXPORT(3, 9)
+int botan_x509_cert_params_builder_add_state(botan_x509_cert_params_builder_t builder, const char* state);
+
+BOTAN_FFI_EXPORT(3, 9)
+int botan_x509_cert_params_builder_add_serial_number(botan_x509_cert_params_builder_t builder,
+                                                     const char* serial_number);
+
+BOTAN_FFI_EXPORT(3, 9)
+int botan_x509_cert_params_builder_add_email(botan_x509_cert_params_builder_t builder, const char* email);
+
+BOTAN_FFI_EXPORT(3, 9)
+int botan_x509_cert_params_builder_add_uri(botan_x509_cert_params_builder_t builder, const char* uri);
+
+BOTAN_FFI_EXPORT(3, 9)
+int botan_x509_cert_params_builder_add_ip(botan_x509_cert_params_builder_t builder, const char* ip);
+
+BOTAN_FFI_EXPORT(3, 9)
+int botan_x509_cert_params_builder_add_dns(botan_x509_cert_params_builder_t builder, const char* dns);
+
+BOTAN_FFI_EXPORT(3, 9)
+int botan_x509_cert_params_builder_add_xmpp(botan_x509_cert_params_builder_t builder, const char* xmpp);
+
+BOTAN_FFI_EXPORT(3, 9)
+int botan_x509_cert_params_builder_add_challenge(botan_x509_cert_params_builder_t builder, const char* challenge);
+
+BOTAN_FFI_EXPORT(3, 9)
+int botan_x509_cert_params_builder_mark_as_ca_key(botan_x509_cert_params_builder_t builder, size_t limit);
+
+BOTAN_FFI_EXPORT(3, 9)
+int botan_x509_cert_params_builder_add_not_before(botan_x509_cert_params_builder_t builder, uint64_t time_since_epoch);
+
+BOTAN_FFI_EXPORT(3, 9)
+int botan_x509_cert_params_builder_add_not_after(botan_x509_cert_params_builder_t builder, uint64_t time_since_epoch);
```

Comment:
Remove these from the params builder - PKCS10 doesn't use them. Instead add not_before/not_after as args to `botan_x509_create_self_signed_cert` similar to `botan_x509_sign_req`